### ISSUE TITLE
saslauthd/ldap_adaptor: failed to register services

### ISSUE THREAD

**Issue by midasi** [grommunio/gromox#99]
We use saslauthd with pam_gromox to authenticate users with the mysql database backend.
Environment is Alpine Linux 3.20 (musl libc).

When we start saslauthd with 3 threads, we are able to authenticate 3 times without issue.
The 4th authentication request fails with "ldap_adaptor: failed to register services".

```
saslauthd[7153] :num_procs  : 3
saslauthd[7153] :mech_option: NULL
saslauthd[7153] :run_path   : /run/saslauthd
saslauthd[7153] :auth_mech  : pam
saslauthd[7153] :using accept lock file: /run/saslauthd/mux.accept
saslauthd[7153] :master pid is: 0
saslauthd[7153] :listening on socket: /run/saslauthd/mux
saslauthd[7153] :using process model
saslauthd[7153] :forked child: 7154
saslauthd[7154] :acquired accept lock
saslauthd[7153] :forked child: 7155
saslauthd[7155] :acquired accept lock
saslauthd[7154] :released accept lock
ldap_adaptor: default host <ldapi:///>, base <>, #conn=8, mailattr=mail
mysql_adaptor: Current schema n131 is recent.
saslauthd[7154] :auth success: [user=johndoe@example.com] [service=smtp] [realm=example.com] [mech=pam]
saslauthd[7154] :response: OK
saslauthd[7155] :released accept lock
saslauthd[7153] :acquired accept lock
ldap_adaptor: default host <ldapi:///>, base <>, #conn=8, mailattr=mail
mysql_adaptor: Current schema n131 is recent.
saslauthd[7155] :auth success: [user=johndoe@example.com] [service=smtp] [realm=example.com] [mech=pam]
saslauthd[7155] :response: OK
saslauthd[7153] :released accept lock
saslauthd[7154] :acquired accept lock
ldap_adaptor: default host <ldapi:///>, base <>, #conn=8, mailattr=mail
mysql_adaptor: Current schema n131 is recent.
saslauthd[7153] :auth success: [user=johndoe@example.com] [service=smtp] [realm=example.com] [mech=pam]
saslauthd[7153] :response: OK
saslauthd[7154] :released accept lock
saslauthd[7155] :acquired accept lock
ldap_adaptor: default host <ldapi:///>, base <>, #conn=8, mailattr=mail
ldap_adaptor: failed to register services
service: init of libgxs_ldap_adaptor.so not successful
saslauthd[7154] :auth failure: [user=johndoe@example.com] [service=smtp] [realm=example.com] [mech=pam] [reason=PAM auth error]
saslauthd[7155] :released accept lock
saslauthd[7153] :acquired accept lock
ldap_adaptor: default host <ldapi:///>, base <>, #conn=8, mailattr=mail
ldap_adaptor: failed to register services
service: init of libgxs_ldap_adaptor.so not successful
saslauthd[7155] :auth failure: [user=johndoe@example.com] [service=smtp] [realm=example.com] [mech=pam] [reason=PAM auth error]

```

**Comment by jengelh** [comment]
That ought to have been fixed by gromox-2.29-39-g6ccf61178 .

**Comment by midasi** [comment]
Thank you! We confirm it's working properly with the mentioned patch.
